src/js/components/history/HistoryList.tsx

```diff
@@ -0,0 +1,56 @@
+import {Map} from 'immutable';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {updateHistoryProperty} from '../../actions/historyActions';
+import {State} from '../../store';
+import {HistoryItem} from './HistoryItem';
+import {MarkRecord} from '../../store/factory/Mark';
+
+const getIn = require('../../util/immutable-utils').getIn;
+
+interface OwnProps {
+}
+interface StateProps {
+  history: any[];
+  groupNames: string[];
+}
+
+interface DispatchProps {
+  updateHistoryProperty: (payload: {property: string, value: any}, id: number) => void;
+}
+
+function mapState(state: State): StateProps {
+  const marks: Map<string, MarkRecord> = state.getIn(['vis', 'present', 'marks']);
+  const groupNames = marks.filter((mark: MarkRecord) => {
+    return mark.type === 'group';
+  }).map((v) => {
+    return v.name.replace(" ", "_");
```

Comment:
i think these names should already be human readable. also, users can rename marks. double check whether this replace might not be necessary